src/operations.rs

```diff
@@ -1,21 +1,79 @@
+            state.registers[source_register_2 as usize],
         );
     }
-    update_flags(destination_resgister, registers);
+    update_flags(destination_register, &mut state.registers);
+}
+/// Load the data from a memory location into the destination register
+/// The number between the () indicates the amount of bits of that field or its value
+/// Instruction: | OP_Code (1010)| DR (3)| PCOffset9 (9)|
+pub(crate) fn load_indirect(instruction: u16, state: &mut State) {
+    let destination_register = instruction >> 9 & 0x7; // Take the 3 DR bits
+    let pc_offset = sign_extend(instruction & 0x1FF, 9); // Take the 9 PCOffset bits and sign_extend them
+    let memory_index = u16::wrapping_add(state.registers[Registers::Rpc], pc_offset);
+    state.registers[destination_register as usize] = state.memory[memory_index as usize];
+    update_flags(destination_register, &mut state.registers);
+}
+/// Binary AND operation with two possible encodings
+/// The number between the () indicates the amount of bits of that field or its value
+/// * Register mode:    |OP_Code (0101)|DR (3)|SR1 (3)|0|00|SR2 (3)|
+/// * Immediate mode:   |OP_Code (0101)|DR (3)|SR1 (3)|1| IMMR5 (5)|
+pub(crate) fn and(instruction: u16, state: &mut State) {
+    let destination_register = (instruction >> 9) & 0x7;
+    let source_register_1 = (instruction >> 6) & 0x7;
+    let mode = (instruction >> 5) & 0x1;
+    if mode == 1 {
+        let value_to_and = sign_extend((instruction) & 0x1F, 5);
+        state.registers[destination_register as usize] =
+            state.registers[source_register_1 as usize] & value_to_and;
+    } else {
+        let source_register_2 = instruction & 0x7;
+        state.registers[destination_register as usize] = state.registers
+            [source_register_1 as usize]
+            & state.registers[source_register_2 as usize];
+    }
+    update_flags(destination_register, &mut state.registers);
+}
+
+/// Conditional branch operator
+/// Instruction: |OP_Code (0000)| n (1)| z (1)| p (1)| PCOffset9 (9)|<br>
+/// The three bits after the OP_Code set which flags will be tested
+/// * n = 1 => The Neg flag is tested
+/// * z = 1 => The Zro flag is tested
+/// * p = 1 => The Pos flag is tested
+///
+/// If the flag tested is has the value 1, then the sign extended PCOffset9 is added to the Program counter<br>
+/// Only one of the flags will have the value 1 at each moment, so if multiple flags are tested only one needs to be in 1 for the branch to occure
+pub(crate) fn conditional_branch(instruction: u16, state: &mut State) {
+    let negative_indicator = (instruction >> 11) & 1;
+    let zero_indicator = (instruction >> 10) & 1;
+    let positive_indicator = (instruction >> 9) & 1;
+    let current_flags = state.registers[Registers::Rcond];
+    if ((negative_indicator & current_flags >> 2) == 1)
+        || ((zero_indicator & current_flags >> 1) == 1)
+        || ((positive_indicator & current_flags) == 1)
```

Comment:
I suggest you split the conditions within the if statement into different variables to make it clearer. Something like:

```rust
let is_<condition1> = (negative_indicator & current_flags) >> 2 == 1;
let is_<condition2> = (zero_indicator) & current_flags) >> 1 == 1;
let is_<condition3> = (positive_indicator & current_flags) == 1;

if is_<condition1> || is_<condition2> || is_<condition3> {
    // Do something...
}
```